### DISCUSSION TITLE
restore certain file

### DISCUSSION THREAD

**Discussion by DannyLi001** [community/community#34275] [New to GitHub] [answered]
How can I restore certain file in a repository to its previous version but holding others the same and then uploaded in github. in other words, can I delete all the commits record to get the original version of a file in github?

**Comment by airtower-luna** [comment, marked as answer]
You can use the [`git checkout <tree-ish> -- <pathspec>…`](https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt-emgitcheckoutem-f--ours--theirs-m--conflictltstylegtlttree-ishgt--ltpathspecgt82308203) form of `git checkout` to set specific file(s) to the state they were at a specific commit (or branch, tag, etc.). Then you can add & commit it as usual.

If you just want to make a new commit with the code changed back to the old version, that's enough. If you really want to modify the commits in between look at [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). Basically you'd need to modify all commits in between that changed the file likewise, and note that changing history can cause a lot of trouble if you have already shared the history with others. :sweat_smile:

**Reply by DannyLi001** [reply]
Thank you

**Reply by airtower-luna** [reply]
You're welcome! Please remember to mark the answer if it solved your problem. :slightly_smiling_face:
